File: trunk/source/add-lib-develop/add-lib-develop/core/src/test/resources/info/scce/addlib/codegenerator/TestProgram.js
function assertEquals(a, b) {
	if (a != b) {
		throw "expected equals but " + a + " != " + b;
	}
}

const predicates = {
	predA: function() {
		return true;
	},
	predB: function() {
		return false;
	},
	predC: function() {
		return true
	}
}

imported = require("./SumOfPredicatesABC.js")

const expected = 2;
const actual = imported.SumOfPredicatesABC.sumOfPredicatesABC(predicates);
assertEquals(expected, actual);

File: trunk/source/add-lib-develop/add-lib-develop/core/src/test/resources/info/scce/addlib/codegenerator/TestProgram.java
public class TestProgram {

    public static void main(String[] args) {
        Predicates p = new Predicates() {

            @Override
            public boolean predA() {
                return true;
            }

            @Override
            public boolean predB() {
                return false;
            }

            @Override
            public boolean predC() {
                return true;
            }
        };
        String result = SumOfPredicatesABC.sumOfPredicatesABC(p);
        String expectedResult = "2.0";
        if (!result.equals(expectedResult)) {
            System.err.println("test program failed: expected " + expectedResult + " but was " + result);
            int exitCodeFailure = 1;
            System.exit(exitCodeFailure);
        } else {
            int exitCodeSuccess = 0;
            System.exit(exitCodeSuccess);
        }
    }
}

